packages/bitfield/src/lib/BitField.ts

```diff
@@ -0,0 +1,279 @@
+				if (value !== (value | 0)) throw new RangeError(`The property "${key}" does not resolve to a safe bitfield value`);
+				if (value <= 0) throw new RangeError(`The property "${key}" resolves to a non-positive value`);
+				mask |= value;
+			}
+
+			this.mask = mask as any;
+		} else {
+			this.zero = 0n as any;
+
+			let mask = 0n;
+			for (const [key, value] of entries) {
+				if (typeof value !== 'bigint') throw new TypeError(`The property "${key}" does not resolve to a bigint`);
+				if (value <= 0n) throw new RangeError(`The property "${key}" resolves to a non-positive value`);
+				mask |= value;
+			}
+
+			this.mask = mask as any;
+		}
+	}
+
+	/**
+	 * Resolves a:
+	 * - `string`: If it's a property of {@link Flags}.
+	 * - `number`: If the BitField processes `number` primitives.
+	 * - `bigint`: If the BitField processes `bigint` primitives.
+	 * - `Array`: Resolves recursively.
+	 * @param resolvable The value to resolve.
+	 * @returns The resolved value.
+	 */
+	public resolve(resolvable: ValueResolvable<this>): ValueType<this> {
+		switch (typeof resolvable) {
+			case 'string':
+				if ((resolvable as string) in this.flags) return this.flags[resolvable as keyof Flags] as any;
+				throw new RangeError('Received a name that could not be resolved to a property of flags');
+			case this.type:
+				return ((resolvable as ValueType<this>) & this.mask) as any;
+			case 'object':
+				if (Array.isArray(resolvable)) return resolvable.reduce((acc, value) => this.resolve(value) | acc, this.zero);
+				throw new TypeError('Received an object value that is not an Array');
+			default:
+				throw new TypeError(`Received a value that is not either type "string", type "${this.type}", or an Array`);
+		}
+	}
+
+	/**
+	 * Checks whether or not `field` contains any of the bits from `bits`.
+	 * @param field The bits to compare the bits from.
+	 * @param bits The bits to compare with.
+	 * @returns Whether or not `field` has any of `bits`'s bits, also denoted as `A ∩ B ≠ ∅`.
+	 */
+	public any(field: ValueResolvable<this>, bits: ValueResolvable<this>): boolean {
+		return (this.resolve(field) & this.resolve(bits)) !== this.zero;
+	}
+
+	/**
+	 * Checks whether or not `field` is a superset of or equal to `bits`.
+	 * @param field The bits to compare the bits from.
+	 * @param bits The bits to compare with.
+	 * @returns Whether or not `field` is a superset of or equal to `bits`, also denoted as `A ⊇ B`.
+	 */
+	public has(field: ValueResolvable<this>, bits: ValueResolvable<this>): boolean {
+		const resolved = this.resolve(bits);
+		return (this.resolve(field) & resolved) === resolved;
+	}
+
+	/**
+	 * Makes the complement of `field`, which is a field of all bits (of `U` or the union of all {@link Flags} bits)
+	 * that do not belong to `A`. It is the result of `U ∖ A`, or `difference(U, field)`.
+	 * @param field The bits to get the complement of.
+	 * @returns The complement of `field`, also denoted `Aᶜ` or `A'`.
```

Comment:
There's only one complement, that's why I called it "the" complement instead of "a" complement.

Reply:
This isn't about singular or plural but about English sentence structure. You always make "a complement". You'll see in articles such as https://www.grammarly.com/blog/complement-compliment/ (which admittedly do not address "the" vs "a" specifically but still apply) and https://dictionary.cambridge.org/dictionary/english/complement that they always write "a complement"

Furthermore the plural of "complement" is "complements" so if this was a matter of pluralization then "the complement" would be wrong as well as it would have to have been "the complements". That is not the case here however.

Reply:
We're talking about mathematical terms here, Wikipedia refers it as "the" complement, 13 times.

https://en.wikipedia.org/wiki/Complement_(set_theory)